packages/plugins/rubygem-smart_proxy_ansible/rubygem-smart_proxy_ansible.spec

```diff
@@ -1,66 +1,98 @@
+Name: %{?scl_prefix}rubygem-%{gem_name}
 Version: 3.0.1
-Release: 1%{?foremandist}%{?dist}
-Group: Applications/System
-License: GPLv3
+Release: 2%{?foremandist}%{?dist}
+Summary: Smart-Proxy Ansible plugin
+Group: Applications/Internet
+License: GPL-3.0
 URL: https://github.com/theforeman/smart_proxy_ansible
 Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
 
+Requires: ansible >= 2.2
 %if 0%{?rhel} == 7
-Requires: tfm-rubygem(smart_proxy_dynflow_core) >= 0.1.5
-Requires: tfm-rubygem(foreman_ansible_core)
+Requires: python-requests
 %else
-Requires: rubygem(smart_proxy_dynflow_core) >= 0.1.5
-Requires: rubygem(foreman_ansible_core)
+Requires: python3-requests
 %endif
-Requires: foreman-proxy >= 1.11.0
-Requires: rubygem(smart_proxy_dynflow) >= 0.1
-Requires: rubygem(smart_proxy_dynflow) < 1.0
 
-Requires: ruby
-Requires: ruby(rubygems)
-
-Requires: ansible >= 2.2
 %if 0%{?rhel} == 7
-Requires: python-requests
+Requires: tfm-rubygem(smart_proxy_dynflow_core) >= 0.1.5
+Requires: tfm-rubygem(foreman_ansible_core)
 %else
-Requires: python3-requests
+Requires: rubygem(smart_proxy_dynflow_core) >= 0.1.5
+Requires: rubygem(foreman_ansible_core)
 %endif
 
-Requires: ruby(release)
-BuildRequires: ruby(release)
-BuildRequires: rubygems-devel
+# start specfile generated dependencies
+Requires: foreman-proxy >= %{foreman_proxy_min_version}
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}ruby
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(smart_proxy_dynflow) >= 0.1
+Requires: %{?scl_prefix}rubygem(smart_proxy_dynflow) < 1
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
 BuildArch: noarch
-
-Provides: rubygem(%{gem_name}) = %{version}
-Provides: foreman-proxy-plugin-ansible
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-proxy-plugin-%{plugin_name} = %{version}
+# end specfile generated dependencies
 
 %description
-Ansible support for Foreman smart proxy
+Smart-Proxy ansible plugin.
+
 
 %package doc
-BuildArch:  noarch
-Requires:   %{name} = %{version}-%{release}
-Summary:    Documentation for rubygem-%{gem_name}
+Summary: Documentation for %{name}
+Group: Documentation
+Requires: %{name} = %{version}-%{release}
+BuildArch: noarch
 
 %description doc
-This package contains documentation for rubygem-%{gem_name}.
+Documentation for %{name}.
 
 %prep
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
 
-%setup -q -c -T
-%gem_install -n %{SOURCE0}
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+%{?scl:scl enable %{scl} - << \EOF}
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
+%{?scl:EOF}
 
 %build
+# Create the gem as gem install only works on a gem file
+%{?scl:scl enable %{scl} - << \EOF}
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+
+if [ -f %{foreman_proxy_dir}/.ansible.cfg ] && [ ! -L %{foreman_proxy_dir}/.ansible.cfg ]; then
```

Comment:
This was done in `%pre` to migrate user systems which might already have the `.ansible.cfg` present. This is needed because the package already ships a symlink and RPM can't deal with that.